driver-core/src/main/com/mongodb/internal/binding/ClusterBinding.java

```diff
@@ -75,20 +77,25 @@ public ReadPreference getReadPreference() {
     }
 
     @Override
-    public ConnectionSource getReadConnectionSource() {
-        return new ClusterBindingConnectionSource(new ReadPreferenceServerSelector(readPreference));
+    public SessionContext getSessionContext() {
```

Comment:
The methods `getSessionContext()` and `getReadConnectionSource()` have been reordered.

Reply:
That was intentional.  I thought the existing order was weird.  But I can change it back if you think it is obfuscatory.

Reply:
If it was intentional, then I have no issue with it.